cpp/ql/src/semmle/code/cpp/models/implementations/Pure.qll

```diff
@@ -30,7 +30,12 @@ class PureStrFunction extends AliasFunction, ArrayFunction, TaintFunction, SideE
         name = "strtol" or
         name = "strtoll" or
         name = "strtoq" or
-        name = "strtoul"
+        name = "strtoul" or
+        name = "wcslen" or
+        name = "_mbslen" or
+        name = "_mbslen_l" or
+        name = "_mbstrlen" or
+        name = "_mbstrlen_l"
```

Comment:
If the underscore-prefixed versions can't be in the `std` namespace, then they should be matched by `hasGlobalName` instead of `hasGlobalOrStdName`.

Reply:
Good point, done.